### DISCUSSION TITLE
Best way to implement Auth

### DISCUSSION THREAD

**Discussion by ArthurDanjou** [nuxt/nuxt#28685] [Questions]
Hello, I am currently building a saas with Nuxt, NuxtHub and supabase (for relational database instead of sqlite).
I am using nuxt-auth-utils with github and google. 
I have to store some informations of the users, as the username, the avatar, the description, etc..., from the data returned by the oauth provider. When is the best moment to upsert (update or create) the user data into the db ? When I receive the data from the oauth provider or at another moment ?

And a second question is, do I have to store this information into a store or I can use the UserSession ?

**Comment by ssw1992** [comment]
I have same question
